app/controllers/customers_controller.rb

```diff
@@ -0,0 +1,89 @@
+class CustomersController < ApplicationController
+  rescue_from ActiveRecord::RecordNotFound, with: :catch_not_found
+
+  before_action :set_customer, only: %i[ show edit update destroy ]
+
+  # GET /customers or /customers.json
+  def index
+    @customers = Customer.all
+  end
+
+  # GET /customers/1 or /customers/1.json
+  def show
+  end
+
+  # GET /customers/new
+  def new
+    @customer = Customer.new
+  end
+
+  # GET /customers/1/edit
+  def edit
+  end
+
+  # POST /customers or /customers.json
+  def create
+    @customer = Customer.new(customer_params)
+    if @customer.save
+      flash.notice = "The customer record was successfully created."
+      redirect_to @customer
+    else
+      render :new, status: :unprocessable_entity
+    end
+    # respond_to do |format|
+    #   if @customer.save
+    #     format.html { redirect_to customer_url(@customer), notice: "Customer was successfully created." }
+    #     format.json { render :show, status: :created, location: @customer }
+    #   else
+    #     format.html { render :new, status: :unprocessable_entity }
+    #     format.json { render json: @customer.errors, status: :unprocessable_entity }
+    #   end
+    # end
```

Comment:
Just some food for thought: Keeping blocks of commented code in production can be useful to reference later, but you can accomplish the same thing by just committing your work and referencing the commit history of your project in github. Not a big deal, but learning to look through git history can be useful!